service/src/main/java/com/bankbazaar/service/controller/TripController.java

```diff
@@ -1,61 +1,73 @@
+import com.bankbazaar.service.mapper.TripMapper;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.ModelAndView;
 
-import javax.validation.Valid;
-import java.util.Optional;
-@RestController
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Controller
 @RequestMapping("/trips")
 public class TripController {
     @Autowired
-    private TripManager service;
+    private TripEntityManager manager;
 
     @Autowired
-    private ModelMapper modelMapper;
-
-    @RequestMapping(method = RequestMethod.POST)
-    public ResponseEntity<TripEntity> addTrip(@Valid @RequestBody TripDto trip) {
+    private TripUserMapEntityManager tripUserMap;
+    @Autowired
+    private TripActivityEntityManager tripActivity;
+    @Autowired
+    private DataManager service;
+    @Autowired
+    private TripMapper modelMapper;
 
-        TripEntity response = service.saveTrip(modelMapper.map(trip, TripEntity.class));
-        return new ResponseEntity<>(response, HttpStatus.OK);
+    @RequestMapping(value = "/create",method = RequestMethod.GET)
+    public String createTrip(){
+        return "create_trip";
     }
 
-    @RequestMapping(method = RequestMethod.GET)
-    public ResponseEntity<TripEntity> findTripById(@RequestParam Long id) {
-
-        Optional<TripEntity> tripData = service.getTripById(id);
-        if (tripData.isEmpty()) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
-
-        return new ResponseEntity<>(tripData.get(), HttpStatus.OK);
-    }
+    @RequestMapping(method = RequestMethod.POST)
+    public String addTrip( @ModelAttribute TripDto trip, Principal principal) {
 
-    @RequestMapping(method = RequestMethod.DELETE)
-    public ResponseEntity<TripEntity> deleteTrip(@RequestParam Long id) {
 
-        if (!service.deleteTrip(id)) {
-            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        }
+        TripEntity response = manager.saveTrip(modelMapper.dtoToDomain(trip));
+        service.addAdmin(response, principal);
 
-        return new ResponseEntity<>(HttpStatus.OK);
+        return "redirect:/trips";
     }
-
-    @RequestMapping(method = RequestMethod.PUT)
-    public ResponseEntity<TripEntity> updateTrip(@Valid @RequestBody TripDto trip) {
-
-        TripEntity response = service.saveTrip(modelMapper.map(trip, TripEntity.class));
-        if(response == null)
+    @RequestMapping(method = RequestMethod.GET)
+    public ModelAndView viewTrip(Principal principal){
+        UserEntity userData = service.userDetails(principal).get();
+        List<TripUserMapEntity> tripList = tripUserMap.getTripsUserId(userData.getUserId());
```

Comment:
Controller should not be having all this logic. This should be in service

Reply:
Changes hasve been made